rendermodules/mesh_lens_correction/LensPointMatches.py

```diff
@@ -0,0 +1,296 @@
+                    (k1xy[:, 0] >= r.min()) &
+                    (k1xy[:, 0] <= r.max()) &
+                    (k1xy[:, 1] >= c.min()) &
+                    (k1xy[:, 1] <= c.max())).flatten()
+            fargs.append([k1xy, k2xy, des1, des2, k1ind])
+            results.append(ransac_chunk(fargs[-1]))
+
+    for result in results:
+        k1 += result[0]
+        k2 += result[1]
+
+    k1 = np.array(k1) / args['downsample_scale']
+    k2 = np.array(k2) / args['downsample_scale']
+    print('found %d matches' % k1.shape[0])
+
+    if k1.shape[0] > args['matchMax']:
+        a = np.arange(k1.shape[0])
+        np.random.shuffle(a)
+        k1 = k1[a[0: args['matchMax']], :]
+        k2 = k2[a[0: args['matchMax']], :]
+        print('  keeping %d' % k1.shape[0])
+
+    #render = renderapi.connect(**args['render'])
+    render = args['render']
+    pm_dict = make_pm(ids, gids, k1, k2)
+    renderapi.pointmatch.import_matches(
+      args['match_collection'],
+      [pm_dict],
+      render=render)
+
+
+def make_pm(ids, gids, k1, k2):
+    pm = {}
+    pm['pId'] = ids[0]
+    pm['qId'] = ids[1]
+    pm['pGroupId'] = gids[0]
+    pm['qGroupId'] = gids[1]
+    pm['matches'] = {'p': [], 'q': [], 'w': []}
+    pm['matches']['p'].append(k1[:, 0].tolist())
+    pm['matches']['p'].append(k1[:, 1].tolist())
+    pm['matches']['q'].append(k2[:, 0].tolist())
+    pm['matches']['q'].append(k2[:, 1].tolist())
+    pm['matches']['w'] = np.ones(k1.shape[0]).tolist()
+    return pm
+
+def parse_tileids(tpjson):
+    tile_ids = np.array(
+                [[m['p']['id'], m['q']['id']]
+                    for m in tpjson['neighborPairs']])
+    
+    unique_ids = np.unique(tile_ids.flatten())
+
+    # determine tile index per tile pair
+    tile_index = np.zeros(
+                    (tile_ids.shape[0], tile_ids.shape[1])
+                    ).astype('int')
+    for i in range(tile_ids.shape[0]):
+        for j in range(tile_ids.shape[1]):
+            tile_index[i,j] = np.argwhere(unique_ids == tile_ids[i,j])
+    
+    return unique_ids, tile_index
+
+
+def load_pairjson(tilepair_file):
+    tpjson = []
+    if tilepair_file is not None:
+        tpjson = json.load(open(tilepair_file, 'r'))
+    return tpjson
+
+
+def load_tilespecs(render, input_stack, unique_ids):
+    tilespecs = []
+    for tid in unique_ids:
+        tilespecs.append(
+            renderapi.tilespec.get_tile_spec_raw(
+                input_stack, tid, render=render))
+    
+    return np.array(tilespecs)
+
+
+def match_image_pairs(render, tilespecs, match_collection, downsample_scale, nfeature_limit, matchMax, tile_index, ncpus=-1, ndiv=1, index_list=None):
+    if ncpus == -1:
+        ncpus = multiprocessing.cpu_count()
+    
+    args = {}
+    args['downsample_scale'] = downsample_scale
+    args['render'] = render 
+    args['match_collection'] = match_collection
+    args['nfeature_limit'] = nfeature_limit
+    args['matchMax'] = matchMax
+
+    pool = multiprocessing.Pool(ncpus)
```

Comment:
Using `WithPool` defined in the renderapi client should make these calls testable without mocking non-forked processes.

Reply:
Yes.. agreed but this and other changes will come in the next version of refactoring without changing the logic of the code... This PR is to just facilitate ease of processing for Tim.